rasterio/crs.pyx

```diff
@@ -825,7 +825,7 @@ cdef class CRS:
         if not value:
             raise CRSError("CRS is empty or invalid: {!r}".format(value))
 
-        elif value.upper().startswith('EPSG:'):
+        elif value.upper().startswith('EPSG:') and "+" not in value:
```

Comment:
My other thought was to remove this entirely as it can be handled by `from_user_input` as well.

Reply:
I think this is fine :+1: